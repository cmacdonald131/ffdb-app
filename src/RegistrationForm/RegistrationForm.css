* {
  box-sizing: border-box;
}

.wrapper {
  background: url(../Images/football_field3.jpg);
  background-size: cover;
  background-position: center center;
  min-height: 100vh;
}

.RegistrationForm {
  margin-left: 30px;
  margin-right: 30px;
}

label {
  color: #ffffff;
  font-size: 18px;
  margin-bottom: 10px;
}

h3 {
  color: #ffffff;
}

.Button {
  margin-bottom: 30px;
}

.name {
  margin-top: 5%;
}

small {
  color: #ffffff;
  text-align: center;
  margin-top: 5px;
}

@media screen and (min-width: 700px) {
  .RegistrationForm {
    margin-right: auto;
  }

small {
  margin-left: 20%;
  
}

  label {
    font-size: 20px;
    margin-left: 150px;
  }

  .Input {
    width: 60%;
    margin-right: auto;
    margin-left: auto;
  }

  .Button {
    margin-left: 150px;
  }
}

@media screen and (min-width: 1100px) {
  label {
    font-size: 20px;
    margin-left: 265px;
  }

  small {
    margin-left: 24%;
    margin-bottom: 30px;
  }

  .Input {
    width: 60%;
    margin-right: auto;
    margin-left: auto;
  }

  .Button {
    margin-left: 265px;
  }

  .name {
    margin-top: 40px;
  }
}
