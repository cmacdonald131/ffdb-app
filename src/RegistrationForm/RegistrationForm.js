import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../Services/auth-api-service'
import Navbar from '../Navbar/Navbar'
import './RegistrationForm.css'

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => { this.props.history.push('/login') }
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { name, email, username, password, confirmPassword } = ev.target

    //confirms that the user input the correct password

    if (password.value !== confirmPassword.value) {
      console.log(password, confirmPassword)
      alert("Passwords don't match!")
    }

    //posts information to server as a new user

    this.setState({ error: null })
    AuthApiService.postUser({
      username: username.value,
      password: password.value,
      name: name.value,
      email: email.value,
    })
      .then(user => {
        name.value = ''
        email.value = ''
        username.value = ''
        password.value = ''
        confirmPassword.value = ''
        console.log(this.props.history.push('/login'))
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <div className="wrapper">
        <Navbar />
        <form
          className='RegistrationForm'
          onSubmit={this.handleSubmit}
        >
          <header>
            <h1>It's time to win</h1>
          </header>
          <div role='alert'>
            {error && <p className='red'>{error}</p>}
          </div>
          <div className='name'>
            <label htmlFor='RegistrationForm__name'>
              Name <Required />
            </label>
            <Input
              name='name'
              type='text'
              placeholder='Enter your name'
              required
              id='RegistrationForm__name'>
            </Input>
          </div>
          <div className='username'>
            <label htmlFor='RegistrationForm__username'>
              Username <Required />
            </label>
            <Input
              name='username'
              type='text'
              placeholder='Enter your Username'
              required
              id='RegistrationForm__username'>
            </Input>
          </div>
          <div className='password'>
            <label htmlFor='RegistrationForm__password'>
              Password <Required />
            </label>
            <Input
              name='password'
              type='password'
              placeholder='Enter your Password'
              required
              id='RegistrationForm__password'>
            </Input>
          </div>
          <div className='confirmPassword'>
            <label htmlFor='RegistrationForm__confirmPassword'>
              Confirm Password <Required />
            </label>
            <Input
              name='confirmPassword'
              type='password'
              placeholder='Confirm your Password'
              required
              id='RegistrationForm__confirmPassword'>
            </Input>
          </div>
          <div className='email'>
            <label htmlFor='RegistrationForm__email'>
              Email <Required />
            </label>
            <Input
              name='email'
              type='text'
              placeholder='Enter your email'
              required
              id='RegistrationForm__email'>
            </Input>
          </div>
          <Button type='submit' className="btn">
            Register
          </Button>
        </form>
      </div>
    )
  }
}
