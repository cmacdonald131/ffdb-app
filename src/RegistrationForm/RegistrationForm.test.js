import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationForm from './RegistrationForm'

describe(`RegistrationForm component`, () => {
  const props = {
    name: 'test-name',
    email: 'test-email',
    username: 'test-username',
    password: 'test-password',
    'data-other': 'test-other-prop'
  }

  it('renders a form.RegistrationForm by default', () => {
    const wrapper = shallow(<RegistrationForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the RegistrationForm given props', () => {
    const wrapper = shallow(<RegistrationForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})