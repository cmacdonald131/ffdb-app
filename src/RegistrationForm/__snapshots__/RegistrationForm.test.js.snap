// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders a form.RegistrationForm by default 1`] = `
<div
  className="wrapper"
>
  <Navbar />
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <header>
      <h1>
        It's time to win
      </h1>
    </header>
    <div
      role="alert"
    />
    <div
      className="name"
    >
      <label
        htmlFor="RegistrationForm__name"
      >
        Name 
        <Required />
      </label>
      <Input
        id="RegistrationForm__name"
        name="name"
        placeholder="Enter your name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="username"
    >
      <label
        htmlFor="RegistrationForm__username"
      >
        Username 
        <Required />
      </label>
      <Input
        id="RegistrationForm__username"
        name="username"
        placeholder="Enter your Username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="RegistrationForm__password"
      >
        Password 
        <Required />
      </label>
      <Input
        id="RegistrationForm__password"
        name="password"
        onClick={[Function]}
        placeholder="Enter your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="confirmPassword"
    >
      <label
        htmlFor="RegistrationForm__confirmPassword"
      >
        Confirm Password 
        <Required />
      </label>
      <Input
        id="RegistrationForm__confirmPassword"
        name="confirmPassword"
        placeholder="Confirm your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="email"
    >
      <label
        htmlFor="RegistrationForm__email"
      >
        Email 
        <Required />
      </label>
      <Input
        id="RegistrationForm__email"
        name="email"
        placeholder="Enter your email"
        required={true}
        type="text"
      />
    </div>
    <Button
      className="btn"
      type="submit"
    >
      Register
    </Button>
  </form>
</div>
`;

exports[`RegistrationForm component renders the RegistrationForm given props 1`] = `
<div
  className="wrapper"
>
  <Navbar />
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <header>
      <h1>
        It's time to win
      </h1>
    </header>
    <div
      role="alert"
    />
    <div
      className="name"
    >
      <label
        htmlFor="RegistrationForm__name"
      >
        Name 
        <Required />
      </label>
      <Input
        id="RegistrationForm__name"
        name="name"
        placeholder="Enter your name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="username"
    >
      <label
        htmlFor="RegistrationForm__username"
      >
        Username 
        <Required />
      </label>
      <Input
        id="RegistrationForm__username"
        name="username"
        placeholder="Enter your Username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="RegistrationForm__password"
      >
        Password 
        <Required />
      </label>
      <Input
        id="RegistrationForm__password"
        name="password"
        onClick={[Function]}
        placeholder="Enter your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="confirmPassword"
    >
      <label
        htmlFor="RegistrationForm__confirmPassword"
      >
        Confirm Password 
        <Required />
      </label>
      <Input
        id="RegistrationForm__confirmPassword"
        name="confirmPassword"
        placeholder="Confirm your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="email"
    >
      <label
        htmlFor="RegistrationForm__email"
      >
        Email 
        <Required />
      </label>
      <Input
        id="RegistrationForm__email"
        name="email"
        placeholder="Enter your email"
        required={true}
        type="text"
      />
    </div>
    <Button
      className="btn"
      type="submit"
    >
      Register
    </Button>
  </form>
</div>
`;
